src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -90,13 +90,12 @@ export class Match {
 	}
 
 	@memoize
-	preview(): { before: string; inside: string; after: string } {
-		let before = this._oneLinePreviewText.substring(0, this._rangeInPreviewText.startColumn - 1),
-			inside = this.getMatchString(),
-			after = this._oneLinePreviewText.substring(this._rangeInPreviewText.endColumn - 1);
+	preview(): { before: string; fullBefore: string; inside: string; after: string } {
+		const fullBefore = this._oneLinePreviewText.substring(0, this._rangeInPreviewText.startColumn - 1),
+			before = lcut(fullBefore, 26, '…');
```

Comment:
I won't block on this... but I wonder why this is done here and not in CSS using `text-overflow`.

Reply:
As a short answer, the reason it's not like that is because it was easier to modify the existing code to add an ellipsis.
Taking a quick skim of the spec for `text-overflow`, it seems that it would be based on an offset based on `px` whereas we  seem to want to measure the offset by character. I might consider changing it to use `text-overflow` in the future, but it's probably not in the scope of this particular PR.

Reply:
`text-overflow` is also for the end of the string; we could abuse RTL text direction to cut from the start but then the ellipsis wouldn't reliably start at the start of the line.